src/AstGen.zig

```diff
@@ -8591,10 +8591,10 @@ fn builtinCall(
         },
 
         .splat => {
-            const len = try expr(gz, scope, .{ .rl = .{ .coerced_ty = .u32_type } }, params[0]);
-            const scalar = try expr(gz, scope, .{ .rl = .none }, params[1]);
+            const scalar = try expr(gz, scope, .{ .rl = .none }, params[0]);
```

Comment:
```suggestion
            const scalar = try comptimeExpr(gz, scope, .{ .rl = .none }, params[0]);
```

Would you mind tacking on this change and adding a behavior test for it?

Behavior test would look something like this:

```zig
test {
    @splat(foo());
}
```

It should make foo() be called at comptime even though the comptime keyword is not used. Same as it already does for e.g. `[foo()]T` (array type syntax).

Reply:
I'm confused, why would you do this? It's perfectly valid for the value being splat-ed to be runtime known; this isn't a length.

Reply:
My mistake, for some reason I thought it was a length, but that's very wrong. I apologize; ignore me!